first_mpi_class/jr.py

```diff
@@ -0,0 +1,205 @@
+import mpi4py as MPI
+import numpy as np
+import sys
+
+class JosueReyes:
+    def point_to_point():    
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+
+        if rank == 0:
+            data = {'a': 7, 'b': 3.14}
+            comm.send(data, dest=1, tag=11)
+        elif rank == 1:
+            data = comm.recv(source=0, tag=11)
+        return True
+
+    def non_blocking():
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+
+        if rank == 0:
+            data = {'a': 7, 'b': 3.14}
+            req = comm.isend(data, dest=1, tag=11)
+            req.wait()
+        elif rank == 1:
+            req = comm.irecv(source=0, tag=11)
+            data = req.wait()
+        return True
+
+    def the_fast_way() -> bool:
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+
+        # passing MPI datatypes explicitly
+        if rank == 0:
+            data = np.arange(1000, dtype='i')
+            comm.Send([data, MPI.INT], dest=1, tag=77)
+        elif rank == 1:
+            data = np.empty(1000, dtype='i')
+            comm.Recv([data, MPI.INT], source=0, tag=77)
+
+        # automatic MPI datatype discovery
+        if rank == 0:
+            data = np.arange(100, dtype=np.float64)
+            comm.Send(data, dest=1, tag=13)
+        elif rank == 1:
+            data = np.empty(100, dtype=np.float64)
+            comm.Recv(data, source=0, tag=13)
+        return True
+
+    def collective_communication():
```

Comment:
Please add the typing, like the other methods:

() -> bool:

Because, this method is returning a Boolean type